Document FIPS image and Openshift default namespace limitation (#6428)
elastic/cloud-on-k8s: docs/advanced-topics/openshift.asciidoc

. Create a namespace to hold the Elastic resources ({eck_resources_list}):
+

> NOTE: A namespace outside of the `default` namespace is required such that default https://docs.openshift.com/container-platform/4.1/authentication/managing-security-context-constraints.html[Security Context Constraint] (SCC) permissions are applied automatically. Elastic resources will not work in the `default` namespace, using the `default` service account.

Reviewer: ```suggestion
NOTE: A namespace other than the `default` namespace is required such that default https://docs.openshift.com/container-platform/4.1/authentication/managing-security-context-constraints.html[Security Context Constraint] (SCC) permissions are applied automatically. Elastic resources will not work in the `default` namespace, using the `default` service account.
```

This is fine but the OpenShift docs are much more prescriptive: 

> You cannot assign a SCC to pods created in one of the default namespaces: default, kube-system, kube-public, openshift-node, openshift-infra, openshift. These namespaces should not be used for running pods or services.

> NOTE: A namespace other than the default namespaces: default, kube-system, kube-public, openshift-node, openshift-infra, openshift is required such that default https://docs.openshift.com/container-platform/4.1/authentication/managing-security-context-constraints.html[Security Context Constraint] (SCC) permissions are applied automatically. Elastic resources will not work properly in any of the default namespaces.

Reviewer 2: Namespaces (a.k.a `projects` in OpenShift) provided by OpenShift starts with `openshift-`. Maybe we should reflect that naming? (see https://docs.openshift.com/online/pro/architecture/core_concepts/projects_and_users.html#architecture-projects-projects-at-install)

Reviewer 3: I'm good with the current version but I find Barkbay's comment interesting, hence the following proposition:
```suggestion
NOTE: A namespace other than the default namespaces (default, kube-*, openshift-*, etc) is required such that default https://docs.openshift.com/container-platform/4.1/authentication/managing-security-context-constraints.html[Security Context Constraint] (SCC) permissions are applied automatically. Elastic resources will not work properly in any of the default namespaces.
```

Reviewer 3: Also, I spotted that the URL uses an old version: https://docs.openshift.com/container-platform/4.1/. We could use the latest https://docs.openshift.com/container-platform/4.12.
This is unrelated to this PR but there is another openshift URL that we could update in `docs/advanced-topics/custom-images.asciidoc`.

Author: Yeah I was going to get @barkbay comments addressed here shortly. I'll check your proposition. Thanks @thbkrkr

Author: @thbkrkr I've updated the wording, using your recommendation, and updating all of the openshift links to v4.12.
